core/src/main/scala/org/apache/spark/ui/storage/StorageTab.scala

```diff
@@ -59,7 +59,15 @@ class StorageListener(storageStatusListener: StorageStatusListener) extends Bloc
 
   override def onStageSubmitted(stageSubmitted: SparkListenerStageSubmitted): Unit = synchronized {
     val rddInfos = stageSubmitted.stageInfo.rddInfos
-    rddInfos.foreach { info => _rddInfoMap.getOrElseUpdate(info.id, info) }
+    rddInfos.foreach { info =>
```

Comment:
OK I'm changing this. I'm not a fan of slower code in favor of brevity (as my impression is that, over time, the entire code base just becomes slow and fixing a couple bottlenecks won't necessarily help things), but it's the in the general spirit of Scala (brevity over speed) so my one optimization would be meaningless anyway.